hack/go-build.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -eu
+
+REPO=github.com/openshift/machine-api-operator
+WHAT=${1:-machine-api-operator}
```

Comment:
Rather fail if $1 is empty in case someone decides to run it just as `./hack/go-build.sh`. Unless that's the intention.

Reply:
that's the intention :)